server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1760,8 +1761,22 @@ public void updateSegmentMetadata(final Set<DataSegment> segments)
           @Override
           public Void inTransaction(Handle handle, TransactionStatus transactionStatus) throws Exception
           {
+            final String updatePayload = StringUtils.format("UPDATE %s SET payload = :payload WHERE id = :id", dbTables.getSegmentsTable());
+            final PreparedBatch batch = handle.prepareBatch(updatePayload);
+
             for (final DataSegment segment : segments) {
```

Comment:
Do we know if the list of input segments to `updateSegmentMetadata` is bounded somewhere? If the number of segments is very large or unbounded, then a single batch transaction for the entire set can be problematic. So we would probably have to break down into multiple batch transactions of reasonable size each. What do you think?

Reply:
No, we should assume that it can be unbounded. We could do something similar to:

https://github.com/apache/druid/blob/efb32810c48fd9c0bf81cefb15179f74a4cde661/server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java#L1460

Reply:
I'm OK with this from a performance standpoint, BUT, it changes the semantics of the code. In the existing code, the delete and update passes are fully committed, OR failed and rolled back as a unit. If we switch to transaction batches, then we lose  the rollback of the whole piece of work and may see partially completed work; in this case we might be better to avoid transactionality altogether since there's no implied atomicity.

Thoughts?

Reply:
Yeah, I agree with the semantics argument. But I took another look at the code and it seems that this method is not even being used anywhere except from a `SegmentMetadataUpdateAction`, which in turn is used by `MoveTask`, `ArchiveTask` and `RestoreTask`, none of which are in use anymore afaik.

@jasonk000 , did you hit a bottleneck with this SQL too or just the DELETE? 

If you did, could you please share more details about the scenario?
If not, we can revert the changes made to this part of the code and just completely remove this code in a later PR.

Reply:
The bottleneck was just with the DELETE, but I noticed the same pattern, so I am happy to revert the update changes. I'll go ahead and do that now (but will leave the TC in).